fem/libceed/solvers-atpmg.cpp

```diff
@@ -84,8 +87,9 @@ int CeedATPMGElemRestriction(int order,
    ierr = CeedElemRestrictionGetNumComponents(er_in, &numcomp); CeedChk(ierr);
    if (numcomp != 1)
    {
-      // todo: this will require more thought
-      return CeedError(ceed, 1, "Not implemented!");
+      // todo: multi-component will require more thought
+      return CeedError(ceed, 1, "Algebraic element restriction not "
+                       "implemented for multiple components.");
```

Comment:
So should I also remove all the `CeedChk(ierr);` checks?

Reply:
I think it's better to do the `CeedChk(ierr)`, I should probably have done it too.

Reply:
For what its worth, `CeedChk()` is in the `ceed-backend.h` header, so its more intended for internal use.

We are revamping the error codes though (https://github.com/CEED/libCEED/pull/691), so it might make sense to pull the libCEED numeric error codes into MFEM's error handling system.

Reply:
I think if it makes sense to use `MFEM_ABORT` instead of `return CeedError`, it also makes sense to change the return type to `void` or an object (as @YohannDudouit asks in other parts of the PR). But if I don't return `int` then `CeedChk` no longer makes sense. So I guess the overall question is how the two error systems interact?

Reply:
How does MFEM normally propagate errors from external libraries?

Reply:
The Petsc interface does this:
```c++
// Error handling
// Prints PETSc's stacktrace and then calls MFEM_ABORT
// We cannot use PETSc's CHKERRQ since it returns a PetscErrorCode
#define PCHKERRQ(obj,err) do {                                                   \
     if ((err))                                                                  \
     {                                                                           \
        PetscError(PetscObjectComm((PetscObject)(obj)),__LINE__,_MFEM_FUNC_NAME, \
                   __FILE__,(err),PETSC_ERROR_REPEAT,NULL);                      \
        MFEM_ABORT("Error in PETSc. See stacktrace above.");                     \
     }                                                                           \
  } while(0);
```
which seems pretty reasonable. I can certainly put together a Ceed-like variant of this.